src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -134,13 +153,16 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
                     // Generate separate queries for each resource type and add them to query list.
                     foreach (string resourceType in _reindexJobRecord.Resources)
                     {
-                        var query = new ReindexJobQueryStatus(resourceType, continuationToken: null)
+                        if (_reindexJobRecord.ResourceCounts[resourceType] > 0)
```

Comment:
In relationship to the TODO comment below on line 488.  If the resource fails to be added to the dictionary, it is not terrible. as long as we don't rely upon it being added successfully.  Perhaps we can check if this key exists first?  If not, or if the count is not 0, then continue with adding the query.

Reply:
Thank you for catching these TODOs, I missed them when putting up my PR! Added a check to see if the resource key does not exist in a8a26fb.